app/core/models.py

```diff
@@ -1,10 +1,73 @@
+import logging
+import json
 from django.db import models
 from model_utils.models import TimeStampedModel
 
+logger = logging.getLogger('dict_config_logger')
+
 
 class SchemaLedger(TimeStampedModel):
     """Model for Uploaded Schemas"""
+    SCHEMA_STATUS_CHOICES = [('published', 'published'),
+                             ('retired', 'retired')]
+
     schema_name = models.CharField(max_length=255)
     schema_iri = models.CharField(max_length=255)
-    schema_metadata = models.FileField(upload_to='schemas/')
-    status = models.CharField(max_length=50)
+    schema_file = models.FileField(upload_to='schemas/',
+                                   null=True,
+                                   blank=True)
+    status = models.CharField(max_length=10,
+                              choices=SCHEMA_STATUS_CHOICES)
+    metadata = models.JSONField(blank=True,
+                                help_text="auto populated from uploaded file")
+    version = models.CharField(max_length=6,
+                               help_text="auto populated from other version "
+                                         "fields")
+    major_version = models.SmallIntegerField()
+    minor_version = models.SmallIntegerField()
+    revision = models.SmallIntegerField()
```

Comment:
If we are following [semantic versioning](https://semver.org/) then we will need a patch version as well -- it looks like that is the intent of the revision field so we could simply rename it.